src/runtime/manifest.ts

```diff
@@ -129,6 +130,20 @@ interface ManifestLoadOptions {
   memoryProvider?: VolatileMemoryProvider;
 }
 
+export class ManifestHandleRetriever implements HandleRetriever {
+  async getHandlesFromManifest(storageKey: string, content: string) : Promise<Handle[]> | null {
+    try {
+      const manifest = await Manifest.parse(content, {});
+      if (manifest.activeRecipe) {
+        return manifest.activeRecipe.handles;
+      }
+    } catch (e) {
+      console.warn(`Error parsing manifest at ${storageKey}:\n${e.message}`);
```

Comment:
are you only passing the storageKey to this method to log the warning? possible this method should throw an error and the caller method should log? not sure.

Reply:
Yes, good suggestion, restructured to throw parse error detail which will fail the promise.

Reply:
Thanks!